scripts/automated_ingestion/eessi_task.py

```diff
@@ -0,0 +1,1201 @@
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "branch '%s' created: '%s'", feature_branch_name, feature_branch)
+        else:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "found existing branch for '%s': '%s'", feature_branch_name, feature_branch)
+
+        pull_request = self._find_pr_for_branch(feature_branch_name)
+        if not pull_request:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "no PR found for branch '%s'", feature_branch_name)
+
+            # TODO: add failure handling (capture result and act on it)
+            self._update_task_states(next_state, default_branch_name, approved_state, feature_branch_name)
+
+            # TODO: add failure handling (capture result and act on it)
+            self._create_pull_request(feature_branch_name, default_branch_name)
+
+            return EESSITaskState.PULL_REQUEST
+        else:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "found existing PR for branch '%s': '%s'", feature_branch_name, pull_request)
+            # TODO: check if PR is open or closed
+            if pull_request.state == "closed":
+                log_message(LoggingScope.TASK_OPS, "INFO",
+                            "PR '%s' is closed, creating issue", pull_request)
+                # TODO: create issue
+                return EESSITaskState.PAYLOAD_STAGED
+            else:
+                log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                            "PR '%s' is open, updating task states", pull_request)
+                # TODO: add failure handling (capture result and act on it)
+                #   THINK about what a failure would mean and what to do about it.
+                self._update_task_states(next_state, default_branch_name, approved_state, feature_branch_name)
+
+                # TODO: add failure handling (capture result and act on it)
+                self._update_pull_request(pull_request)
+
+                return EESSITaskState.PULL_REQUEST
+
+    @log_function_entry_exit()
+    def _handle_add_pull_request(self):
+        """Handler for ADD action in PULL_REQUEST state"""
+        log_message(LoggingScope.TASK_OPS, "INFO", "Handling ADD action in PULL_REQUEST state: '%s'",
+                    self.description.get_task_file_name())
+        # Implementation for adding in PULL_REQUEST state
+        # we got here because the state of the task is PULL_REQUEST in the default branch
+        # determine branch and PR and state of PR
+        # PR is open --> just return EESSITaskState.PULL_REQUEST
+        # PR is closed & merged --> deployment is approved
+        # PR is closed & not merged --> deployment is rejected
+        feature_branch_name = self._determine_feature_branch_name()
+        # TODO: check if feature branch exists, for now ASSUME it does
```

Comment:
```suggestion
```

I think this comment can be removed. _find_pr_for_branch returns none if the PR doesn't exist, which is handled gracefully by the rest of _handle_add_pull_request. So I don't think there is anything 'TODO' here?